src/silx/resources/__init__.py

```diff
@@ -258,9 +261,17 @@ def _resource_filename(
         return resource_path
 
     package_name = resource_directory.package_name
+
+    cache_key = package_name, resource_name
+    cached_path = _file_cache.get(cache_key, None)
+    if cached_path is not None:
+        return cached_path
+
     ref = importlib.resources.files(package_name) / resource_name
-    path = _FILE_MANAGER.enter_context(importlib.resources.as_file(ref))
-    return str(path.absolute())
+    path = _file_manager.enter_context(importlib.resources.as_file(ref))
+    path_string = str(path.absolute())
+    _file_cache[cache_key] = path_string
+    return path_string
```

Comment:
question: is using `functools.cache` wouldn't simplify things ? something like:

```
def _resource_filename(
    resource: str, default_directory: Optional[str] = None
) -> str:
    resource_directory, resource_name = _get_package_and_resource(resource,
                                                                  default_directory=default_directory)

    if resource_directory.forced_path is not None:
        # if set, use this directory
        base_dir = resource_directory.forced_path
        resource_path = os.path.join(base_dir, *resource_name.split('/'))
        return resource_path

    package_name = resource_directory.package_name
    return  _get_cached_path(package_name, resource_name)

@cache
def _get_cached_path(package_name, resource_name):
       if sys.version_info < (3, 9):
        return pkg_resources.resource_filename(package_name, resource_name)

    file_context = importlib.resources.as_file(
        importlib.resources.files(package_name) / resource_name)
    path = _file_manager.enter_context(file_context)
    path_string = str(path.absolute())
    return path_string
```

Reply:
Yes! I'll use it

Reply:
Done